app/services/activitypub/fetch_remote_status_service.rb

```diff
@@ -80,4 +80,20 @@ def expected_activity_type?
   def expected_object_type?
     equals_or_includes_any?(@json['type'], ActivityPub::Activity::Create::SUPPORTED_TYPES + ActivityPub::Activity::Create::CONVERTED_TYPES)
   end
+
+  def fetch_status(uri, id_is_known, on_behalf_of = nil)
+    begin
+      fetch_resource(uri, id_is_known, on_behalf_of, true)
+    rescue Mastodon::UnexpectedResponseError => e
+      return unless e.response.code == 404
+
+      # If this is a 404 from a status from an account that has no local followers, delete it
+      existing_status = Status.find_by(uri: uri)
```

Comment:
this seemed a little expensive to load the whole status and use the instance attribute, rather than using the scope version of `unsubscribed` and testing if the count was nonzero, but i figure since we're touching this status it might be in the cache.